packages/actor-extract-links-extract-tree/lib/ActorExtractLinksTree.ts

```diff
@@ -70,15 +91,43 @@ export class ActorExtractLinksTree extends ActorExtractLinks {
   private getTreeQuadsRawRelations(
     quad: RDF.Quad,
     url: string,
-    pageRelationNodes: Set<string>,
+    pageRelationNodes: Map<string, string>,
     nodeLinks: [string, string][],
+    rootNodeEffectiveSubject: Set<string>,
   ): void {
-    // If it's a relation of the current node
-    if (quad.subject.value === url && quad.predicate.equals(ActorExtractLinksTree.aRelation)) {
-      pageRelationNodes.add(quad.object.value);
-      // If it's a node forward
-    } else if (quad.predicate.equals(ActorExtractLinksTree.aNodeType)) {
+    if (
+      (this.looseMode || !this.looseMode && quad.subject.value === url) &&
+      (quad.predicate.equals(ActorExtractLinksTree.aView) ||
+      quad.predicate.equals(ActorExtractLinksTree.aSubset))) {
+      rootNodeEffectiveSubject.add(quad.object.value);
+    }
+
+    if (
+      (this.looseMode || !this.looseMode && quad.object.value === url) &&
+    quad.predicate.equals(ActorExtractLinksTree.isPartOf)) {
+      rootNodeEffectiveSubject.add(quad.subject.value);
+    }
+
+    if (quad.predicate.equals(ActorExtractLinksTree.aRelation)) {
+      // If it's a relation of the current node
+      pageRelationNodes.set(quad.object.value, quad.subject.value);
+    }
+
+    // If it's a node forward
+    if (quad.predicate.equals(ActorExtractLinksTree.aNodeType)) {
       nodeLinks.push([ quad.subject.value, quad.object.value ]);
     }
   }
 }
+
+export interface IActorExtractLinksTreeArgs
+  extends IActorArgs<IActionExtractLinks, IActorTest, IActorExtractLinksOutput> {
+  /**
+   * If true, then during the traversal of TREE document
```

Comment:
I would suggest to simplify this whole block to the following (assuming `strict`):
`If relationships should strictly correspond to the current document's URL.`